model.py

```diff
@@ -9560,7 +9549,16 @@ def featured_works(self):
         return Work.from_identifiers(_db, identifiers)
 
     def add_entry(self, work_or_edition, annotation=None, first_appearance=None,
-                  featured=None):
+                  featured=None, update_external_index=True):
+        """Add a work to a CustomList.
+
+        :param update_external_index: When a Work is added to a list,
+        its external index needs to be updated. The only reason not to
+        do this is when the current database session already contains
+        a new WorkCoverageRecord for this purpose (e.g. because the
+        Work was just created) and creating another one would violate
+        the workcoveragerecords table's unique constraint.
+        """
```

Comment:
Not really happy about this particular change but 'automatically add every new work to a custom list' seems like a magnet for these problems.

Reply:
I don't see why `Work.external_index_needs_updating` isn't handling this case. Doesn't it return the existing coverage record if one already exists?

Reply:
It doesn't -- it tries to create a second new coverage record, and when the session is flushed, there's an IntegrityError. You can duplicate this by removing the `and update_internal_index` bit and running `nosetests tests/test_model.py:TestCollection.test_custom_lists`.

Behind the scenes the duplicate coverage records are being created by `get_one_or_create`. So the creation of the second record is succeeding inside the nested subtransaction, but then when it's time to commit for real there's a problem.